x-pack/elastic-agent/pkg/agent/transpiler/ast_test.go

```diff
@@ -704,6 +706,123 @@ func TestSelector(t *testing.T) {
 	}
 }
 
+func TestAST_Apply(t *testing.T) {
```

Comment:
Some tests with golden files could be useful here to make it easier to follow.

Reply:
Yes that would have made it easier to read.